transformer_lens/utils.py

```diff
@@ -576,4 +576,13 @@ def get_dataset(dataset_name: str) -> Dataset:
         dataset = load_dataset("NeelNanda/wiki-10k", split="train")
     else:
         raise ValueError(f"Dataset {dataset_name} not supported")
-    return dataset
\ No newline at end of file
+    return dataset
+
+def get_device_for_block_index(index: int, cfg: HookedTransformerConfig, device: Optional[Union[torch.device, str]] = None):
+    assert cfg.device is not None
+    assert cfg.layers_per_device is not None
+    if device is None:
+        device = cfg.device
+    if isinstance(device, torch.device):
+        device = device.type
+    return torch.device(device, index // cfg.layers_per_device)
```

Comment:
Nice! Yeah something like this (although it doesn't exactly have to be this - just whatever makes sense for testing). I think we can just calculate layers_per_device in this function as well. 

Btw - I think we can move this to its own file rather than continuing this giant one. Bit of a shame we have a utils file rather than folder, but maybe put it in say `transformer_lens/utilities/devices.py`? It's then worth adding a docstring/comments as well.